iib/workers/dogpile_cache.py

```diff
@@ -0,0 +1,75 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+import os
+import threading
+
+from celery import current_task
+from dogpile.cache import make_region
+
+from iib.workers.config import get_worker_config
+
+
+def generate_cache_key(fname, *args, **kwargs):
```

Comment:
If you use my suggestion of using `cache_on_arguments`, I believe the default function to generate the cache key will work and this custom function would not be needed.

Reply:
Thank you, I changed the code, could you have a look please?